### PR TITLE
Added support for babysteps to show number of steps applied

### PR DESCRIPTION
**Pull request by atunguyd**
This is a pull request generated from my issue raised entitled "Request: Feedback for baby steps #283" 

I am now familiar with github so please excuse me if this has been done incorrectly.

Code provided adds support to display the babysteps being applied on the LCD panel in units of steps.

Please note that I have only compiled this on my pre RC2 build, I am not yet able to get RC2 and beyond to compile at all so I am unable to test if these changes actually do compile.

### CODE REVIEW

**Comment by Wackerbarth** [comment]
"I am not yet able to get RC2 and beyond to compile at all" -- This is an issue that you need to address urgently. We, the Marlin developers and testers, have far too few resources to "experiment" with code that the author has not actually tested.

What particular difficulties are you encountering in your attempt to compile the current development tree?
Perhaps you should open an ISSUE for the topic. I suspect that if you are having trouble, any assistance provided will also be helpful to others who are in a similar situation.

Also, with respect to pull requests, you should ALWAYS create a new branch (the same as master, but with a different name) to which you apply your changes. Otherwise, you will encounter extreme difficulty in keeping your work in sync with other changes to the development tree.

**Comment by atunguyd** [comment]
I was merely providing a change as part of the spirit of the open source develoment. 

If you do not want the change or are too busy to bring it in then that is fine, I will be in no way offended if you ignore it.

I did try to get the RC2 to compile (without my changes) but this seems to be poorly documented or I am too much of a fool to figure it out. The readme.md file doesn't really have step by step instructions on how to compile the new RC2 release, and it refers to https://github.com/MarlinFirmware/Marlin/wiki for documentation which in turn refers you to the github repo so it is somewhat cyclic.

While I understand that the developers here are busy, so too are others, I have done what I can to implement this and it is working fine for me, in all honesty the changes are very minor and I dont see   them being any different in the RC2 build than that of RC1, I would expect them to compile and work the same, it is just that I have not tested this.

Apologies for not creating a new branch, I did however say that this is the first time I have used github, and coming from svn there are some differences that I would need to get my head around. 

If you look at the issue that I raised and referred to here, I had provided my source changes in there, I then took the time to figure out what I could about getting git to work and provide a pull request. 

I feel that I have done all I can, I opened an issue and then proceeded to fix it myself and provide the fix to you, I believe this is how the spirit of open source should work, if what I have done has violated this in some way then I apologize.

**Comment by Wackerbarth** [comment]
@atunguyd -- Thanks for your participation. I appreciate your effort to contribute and the explanation which you have given. There is nothing for which you need to apologize.

You do raise an important issue concerning the wiki. Let me point out that, until recently, what little documentation that existed was on another site to which the developers were unable to obtain access. The one developer who did have access seems to have no interest in the documentation.

I am concerned that you are unable to transition from RC1 to RC2 (and actually RCBugFix which will be a part of RC3). The incremental changes along that path are few and I have pushed hard to delay additions which would require retrofitting. I would really encourage you to upgrade. And, if there are any difficulties in doing so, we need to know of them so that we might improve things.

**Comment by atunguyd** [comment]
To be honest I only gave a small amount of time towards trying to get it to compile with RC2 because I only had a small window whilst at work here.

I did not raise an issue as I feared that there might already be a document of some sort and I would have preferred raising an issue only to be told to RTFM.

If there is no documentation then maybe this needs to be put together an placed as a readme in the gi repo.

I downloaded arduino 1.6.6 and installed it. I then copied the marlin source into C:\Program Files (x86)\Arduino\libraries\marlin\ and then opened the ino file at C:\Program Files (x86)\Arduino\libraries\marlin\examples\MarlinFirmware\MarlinFirmware.ino
I then tried to build and I hit the error:
#error Compilation using cores other than those provided in this distribution are unsupported
I fiddled around with trying to use the default configuration in the same folder as the ino file with no success and then ran out of time. 
I would need to try again at a later date.

None the less I think a step by step text file on how to get and compile the RC2 build would be in order.

**Comment by Wackerbarth** [comment]
OK, we have a miscommunication here. "RC2" is somewhat different from "dev" (or "master"). In fact, it is in a different version of the repository. The build procedure is the major change between the two.

As for Documentation, I know that it is very sparse. However, there is some in the MarlinDev/wiki and the update to reference it is awaiting merge approval.

Fundamentally, as the error indicates, you are not using the proper "core". You will find instructions for that installation at https://github.com/MarlinFirmware/MarlinDev/wiki/Setting-up-the-Arduino-IDE.

**Comment by atunguyd** [comment]
Apologies - I assumed that dev trunk was the candidate for RC2

**Comment by Wackerbarth** [comment]
No, RC2 already exists and RCBugFix is the work-in-progress that will become RC3. These are all Release Candidates for the Marlin 1.1 series.

"dev", etc. is the development work that will become Marlin 1.2. However, at the present time, the differences between these series are mostly in the structure of the files rather than the functionality that actually ends up in the firmware.  Certainly, that will change over time.

**Comment by atunguyd** [comment]
Ok, so I got around to getting the dev branch to build with my install, I think the https://github.com/MarlinFirmware/MarlinDev/wiki/Setting-up-the-Arduino-IDE should be in a readme somewhere in the source.

I can confirm that my changes compile correctly now, one comment though, the BABYSTEPPING define needs a few other options enabled and this is not checked for, so simply enabling BABYSTEPPING by itself can break the build.

**Comment by Wackerbarth** [comment]
@atunguyd -- Thanks for your comments. I agree that we need better references to the (little) documentation that we have. In my defense, I will note that it has been only VERY recently that this wiki was created and the corresponding updates would be appropriate.

Perhaps you would be willing to propose a revision to the README, etc.

Also, re BABYSTEPS, we have a section of code called SanityChecks.h . Again, proposed additions would be welcome.

File: ultralcd.cpp [hunk 1]
```diff
@@ -458,14 +458,23 @@ void lcd_set_home_offsets() {
...
       babystepsTodo[axis] += BABYSTEP_MULTIPLICATOR * (int)encoderPosition;
+      BabyStepValue[axis] += BABYSTEP_MULTIPLICATOR * (int)encoderPosition;
+      sprintf_P(value,PSTR("%d"),BabyStepValue[axis]);
       encoderPosition = 0;
       lcdDrawUpdate = 1;
+    }else{
+	   sprintf_P(value,PSTR("%d"),BabyStepValue[axis]);
     }
-    if (lcdDrawUpdate) lcd_implementation_drawedit(msg, "");
+    if (lcdDrawUpdate) lcd_implementation_drawedit(msg,value);
```
Comment (thinkyhead) [inline ultralcd.cpp]: No need to make a string out of `BabyStepValue[axis]`. Just do this instead:
```cpp
if (lcdDrawUpdate) lcd_implementation_drawedit(msg, itostr3(BabyStepValue[axis]));
```
Variables in Marlin should start with a lower-case letter. Camel Case is fine: `babyStepValue`.

Babystepping is quite a simple feature, so there's no need to set array values to 0 with a function `BabyStepsHomeAxis`. We could just use `babyStepValue[axis] = 0;` if needed.

**Comment by thinkyhead** [comment]
The `RCBugFix` code just got this enhancement. It only displays the number of babysteps for the current babystep command. The value resets to zero when you exit the babystep command. The `RCBugFix` code will be getting migrated to MarlinDev before too long, so this can be closed.

**Pull request closed without merging**